### PR TITLE
[feat] windows golang update

### PR DESCRIPTION
**Pull request by bragi92**
[comment]: # (Note that your PR title should follow the conventional commit format: https://conventionalcommits.org/en/v1.0.0/#summary)
# PR Description

[comment]: # (The below checklist is for PRs adding new features. If a box is not checked, add a reason why it's not needed.)
# New Feature Checklist

- [ ] List telemetry added about the feature.
- [ ] Link to the one-pager about the feature.
- [ ] List any tasks necessary for release (3P docs, AKS RP chart changes, etc.) after merging the PR.
- [ ] Attach results of scale and perf testing.

[comment]: # (The below checklist is for code changes. Not all boxes necessarily need to be checked. Build, doc, and template changes do not need to fill out the checklist.)
# Tests Checklist

- [ ] Have end-to-end Ginkgo tests been run on your cluster and passed? To bootstrap your cluster to run the tests, follow [these instructions](/otelcollector/test/README.md#bootstrap-a-dev-cluster-to-run-ginkgo-tests).
  - Labels used when running the tests on your cluster:
    - [ ] `operator`
    - [ ] `windows`
    - [ ] `arm64`
    - [ ] `arc-extension`
    - [ ] `fips`
- [ ] Have new tests been added? For features, have tests been added for this feature? For fixes, is there a test that could have caught this issue and could validate that the fix works?
  - [ ] Is a new scrape job needed?
    - [ ] The scrape job was added to the folder [test-cluster-yamls](/otelcollector/test/test-cluster-yamls/) in the correct configmap or as a CR. 
  - [ ] Was a new test label added?
    - [ ] A string constant for the label was added to [constants.go](/otelcollector/test/utils/constants.go).
    - [ ] The label and description was added to the [test README](/otelcollector/test/README.md).
    - [ ] The label was added to this [PR checklist](/.github/pull_request_template).
    - [ ] The label was added as needed to [testkube-test-crs.yaml](/otelcollector/test/testkube/testkube-test-crs.yaml).
  - [ ] Are additional API server permissions needed for the new tests?
    - [ ] These permissions have been added to [api-server-permissions.yaml](/otelcollector/test/testkube/api-server-permissions.yaml).
  - [ ] Was a new test suite (a new folder under `/tests`) added?
    - [ ] The new test suite is included in [testkube-test-crs.yaml](/otelcollector/test/testkube/testkube-test-crs.yaml).

### CODE REVIEW

File: .pipelines/azure-pipeline-build.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ trigger:
   branches:
     include:
     - main
-
+    - kaveesh/golang_windows
```
Comment (bragi92) [inline .pipelines/azure-pipeline-build.yml:5]: remove

File: otelcollector/build/windows/Dockerfile [hunk 2]
```diff
@@ -36,40 +35,15 @@ COPY ./telegraf/telegraf-prometheus-collector-windows.conf $tmpdir/telegraf/
...
-RUN choco install -y ruby --version 2.6.5.1 --params "'/InstallDir:C:\ruby26'" \
-&& choco install -y msys2 --version 20211130.0.0 --params "'/NoPath /NoUpdate /InstallDir:C:\ruby26\msys64'" \
-&& choco install -y vim
-
-# gangams - optional MSYS2 update via ridk failing in merged docker file so skipping that since we dont need optional update
-RUN refreshenv \
-&& ridk install 3 \
-&& echo gem: --no-document >> C:\ProgramData\gemrc \
-&& gem install cool.io -v 1.5.4 --platform ruby \
-&& gem install oj -v 3.3.10 \
```
Comment (bragi92) [inline otelcollector/build/windows/Dockerfile:57]: Clean up old ruby files

File: otelcollector/build/windows/Dockerfile [hunk 3]
```diff
@@ -36,40 +35,15 @@ COPY ./telegraf/telegraf-prometheus-collector-windows.conf $tmpdir/telegraf/
...
-&& gem install fluentd -v 1.12.2 \
-&& gem install win32-service -v 1.0.1 \
-&& gem install win32-ipc -v 0.7.0 \
-&& gem install win32-event -v 0.6.3 \
-&& gem install windows-pr -v 1.2.6 \
-&& gem install tomlrb -v 1.3.0 \
-&& gem install deep_merge -v 1.2.1\
-&& gem install colorize\
-&& gem sources --clear-all
-
```
Comment (bragi92) [inline otelcollector/build/windows/Dockerfile:68]: Do macro based switching for process_utilities for linux versus windows build

File: otelcollector/build/windows/scripts/main.ps1 [hunk 4]
```diff
@@ -488,4 +488,4 @@ Set-Content -Path /opt/microsoft/liveness/azmon-container-start-time $epochTimeN
 
 # Notepad.exe | Out-Null
 Write-Output "Starting ping to keep the container running"
-ping -t 127.0.0.1 | Out-Null
```
Comment (bragi92) [inline otelcollector/build/windows/scripts/main.ps1:491]: remove file.

File: otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/local_testing_aks.ps1 [hunk 5]
```diff
@@ -6,9 +6,9 @@
 # Chart-template.yaml file.
 
 # Define variables
-$ImageTag = "6.8.14-kaveesh-q-07-19-2024-e7626ca2"
-$AKSRegion = "westeurope"
```
Comment (bragi92) [inline otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/local_testing_aks.ps1:10]: Add this in the backdoor deploy readme instructions.

File: otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/ama-metrics-daemonset.yaml [hunk 6]
```diff
@@ -438,14 +438,12 @@ spec:
...
-                - cmd
-                - /c
-                - C:\opt\microsoft\scripts\livenessprobe.cmd
-            periodSeconds: 60
-            initialDelaySeconds: 300
-            timeoutSeconds: 60
+            httpGet:
+              path: /health
+              port: 8080
+            initialDelaySeconds: 60
```
Comment (bragi92) [inline otelcollector/deploy/addon-chart/azure-monitor-metrics-addon/templates/ama-metrics-daemonset.yaml:444]: initialDelaySeconds : 300

File: otelcollector/main/main.go [hunk 7]
```diff
@@ -67,32 +76,39 @@ func main() {
...
 	} else {
 		shared.StartMdsdForUnderlay()
 	}
 
-	// update this to use color coding
-	shared.PrintMdsdVersion()
+	if osType == "linux" {
+		// update this to use color coding
+		shared.PrintMdsdVersion()
+	}
```
Comment (bragi92) [inline otelcollector/main/main.go:111]: print version for windows too

File: internal/grafana_uami/action.ps1 [hunk 8]
```diff
@@ -0,0 +1,10 @@
+# ARMClient doc: https://github.com/projectkudu/ARMClient
+# ARMClient login
+
+$grafanaResourceId="/subscriptions/{sub_id}/resourceGroups/{rg_name}/providers/Microsoft.Dashboard/grafana/{name}"
+$grafanaApiVersion="2023-10-01-preview"
+
```
Comment (bragi92) [inline internal/grafana_uami/action.ps1:6]: Do fair comparison between performance of older bits versus newer bits

File: otelcollector/main/main.go (hunk 7 above)
Comment (bragi92) [reply otelcollector/main/main.go:111]: created a task for this.

File: otelcollector/build/windows/Dockerfile (hunk 3 above)
Comment (bragi92) [reply otelcollector/build/windows/Dockerfile:68]: Golang doesn't have the same macro based switching like ruby. We have to create two separate files from what I can see so I'm going forward with the docker ignore method here

File: internal/grafana_uami/action.ps1 (hunk 8 above)
Comment (bragi92) [reply internal/grafana_uami/action.ps1:6]: Will do this as part of the merge in the dev cluster for fair comparison.

**Review by vishiy — APPROVED** [review]

**Pull request merged**